typings/index.d.ts

```diff
@@ -1527,6 +1526,7 @@ declare module 'discord.js' {
     ): Promise<Webhook>;
     public setNSFW(nsfw: boolean, reason?: string): Promise<TextChannel>;
     public setRateLimitPerUser(rateLimitPerUser: number, reason?: string): Promise<TextChannel>;
+    public setTopic(topic: string, reason?: string): Promise<TextChannel>;
```

Comment:
```suggestion
    public setTopic(topic: string, reason?: string): Promise<this>;
```
If the methods is called on `NewsChannel` you will have the correct returned type with `this`.

Reply:
Fixed. Thank you!

Reply:
Isn't this broken on all methods then?